pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -877,7 +877,7 @@ public class Utils {
         } finally {
             bc.abort();
             // Roll back and use the same action tracking as last build, if any.
-            if (previousAction != null) {
+            if (r.getAction(DeclarativeJobPropertyTrackerAction.class) == null && previousAction != null) {
```

Comment:
Test added - making sure we don't have multiple `DeclarativeJobPropertyTrackerAction`s.